targets/frontend/src/pages/glossary/index.js

```diff
@@ -0,0 +1,232 @@
+import { IoMdAdd, IoMdCloseCircleOutline } from "react-icons/io";
+import { Button, IconButton } from "src/components/button";
+import { Layout } from "src/components/layout/auth.layout";
+import { withCustomUrqlClient } from "src/hoc/CustomUrqlClient";
+import { withUserProvider } from "src/hoc/UserProvider";
+import { useDebouncedState } from "src/hooks/";
+import { useUser } from "src/hooks/useUser";
+import { Flex, Input, jsx, Label, Spinner } from "theme-ui";
+import { useQuery } from "urql";
+
+const getGlossaryQuery = `
+  query getGlossary {
+    glossary(order_by: {term: asc}) {
+      term
+      id
+    }
+  }
+`;
+
+const context = { additionalTypenames: ["glossary"] };
+
+export function GlossaryPage() {
+  const { isAdmin } = useUser();
+  const [search, setSearch] = useState("");
+  const [
+    displayedTerms,
+    setDisplayedTerms,
+    setDebouncedDisplayedTerms,
+  ] = useDebouncedState(undefined);
+  const [isSearching, setSearching] = useState(false);
+  const [{ isFetching, data: { glossary = [] } = {} }] = useQuery({
+    context,
+    query: getGlossaryQuery,
+  });
+
+  useEffect(() => {
+    // prevent loader at first display of page
+    if (!displayedTerms && !search) return setDisplayedTerms(glossary);
+    setSearching(true);
+    setDebouncedDisplayedTerms(
+      glossary.filter((entry) =>
+        entry.term.toLowerCase().includes(search.toLowerCase().trim())
+      )
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [glossary, search, setDisplayedTerms, setDebouncedDisplayedTerms]);
+
+  useEffect(() => {
+    setSearching(false);
+  }, [displayedTerms]);
+
+  const termsByLetters = useMemo(() => getTermsByLetter(displayedTerms), [
+    displayedTerms,
+  ]);
+
+  return (
+    <Layout title={`Glossaire`}>
+      {isFetching ? (
+        <Spinner />
+      ) : (
+        <>
+          <Flex
+            sx={{
+              alignItems: "center",
+              justifyContent: "space-between",
+            }}
+          >
+            <ul
+              sx={{
+                display: "flex",
+                flex: "1 1 auto",
+                flexWrap: "wrap",
+                listStyleType: "none",
+                m: 0,
+                mr: "xsmall",
+                p: 0,
+              }}
+            >
+              {termsByLetters.map(({ letter, terms }) => (
+                <li key={`letter-${letter}`} sx={{ fontSize: "large" }}>
+                  {terms.length > 0 ? (
+                    <a
+                      href={`#ancre-${letter}`}
+                      sx={{ padding: "xxsmall", ...linkStyles }}
+                    >
+                      {letter}
+                    </a>
+                  ) : (
+                    <span sx={{ color: "muted", padding: "xxsmall" }}>
+                      {letter}
+                    </span>
+                  )}
+                </li>
+              ))}
+            </ul>
+            {isAdmin && <AddATermButton />}
```

Comment:
I think auth user can should update glossary

Reply:
Is this also the case for the themes ?

Reply:
we should probably have a meeting with florent to decide what a user profile should do